src/core/audio/AudioPlayer.cpp

```diff
@@ -9,6 +9,13 @@
 
 class Settings;
 
+AudioPlayer::AudioPlayer(Control& control, Settings& settings):
+        control(control),
+        settings(settings),
+        audioQueue(std::make_unique<AudioQueue<float>>()),
+        portAudioConsumer(std::make_unique<PortAudioConsumer>(*this, *audioQueue)),
+        vorbisProducer(std::make_unique<VorbisProducer>(*audioQueue)) {}
```

Comment:
Question: Does such refactorings only happen to classes that have a non default destructor?

Reply:
Yes. Only AudioPlayer, AudioRecorder and AudioController. Previously, the unique_ptr's were initialized in the header file.
Those classes can not have default constructor, because of the references to Settings and Control.

Reply:
I talked from destructors which are relevant.

Reply:
Oh sorry, I misread. Actually, AudioController has a default destructor indeed (the other two don't).

> Classes having a implicitly default destructor should not be refactored in such a way.

Why is that an issue?

Anyway, I can reverse this last commit.

Reply:
No that's fine. There is a tradeoff: compile time vs runtime. By moving the destructor into the source the compiler can't see anymore, that the destructor is trivial, preventing optimisations. Also then, the - rule of 5 applies. But since std::unique_ptr break Copy ctors and all ctors are deleted this is fine.

Reply:
Btw you can merge this. I just wanted to make sure, that this sort of changes should be decided by hand.

Reply:
Thanks for the explanation @Febbe!